### PR TITLE
Group common logic for browser and node envs

### LINKED ISSUE #18: Common part for nodejs and browser implementations
As codebase grows it would be great to extract common code for nodejs and browser implementations

### PR DESCRIPTION
**Pull request by rustworthy**
- [x] tests and linter show no problems (`npm t`)
- [ ] tests are added/updated for bug fixes and new features
- [x] code is properly formatted (`npm run fmt`)
- [x] description of changes is added in CHANGELOG.md
- [x] update .d.ts typings

Closes: https://github.com/tshemsedinov/wasm-import/issues/18

### CODE REVIEW

**Comment by rustworthy** [comment]
@tshemsedinov 

Conceptually, what do you make of this approach?
image

**Comment by rustworthy** [comment]
I remember you turned down the class-based approach, but I still believe it is worth considering. We could group the common logic in the base class and implement 'compile' method in the sub classes, or use an open constructor. E.g. compile(path) {wa.compile streaming(resp)} for browser, and compile(path) {wa.compile(buffer)} for node

**Comment by rustworthy** [comment]
Issue with rust binary can be reproduced locally with  rustc 1.73. Will figure it out.

image

**Comment by rustworthy** [comment]
> Issue with rust binary can be reproduced locally with rustc 1.73. Will figure it out.
> 
> image

Fixed with:
https://github.com/tshemsedinov/wasm-import/pull/25

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
please rename lib/loader to lib/node to see diff

File: lib/node.js [hunk 1]
```diff
@@ -51,14 +51,33 @@ const prepareExports = (instance) => {
...
+const compileInServerEnv = async (fileName) => {
   const buffer = await fsp.readFile(fileName);
   const compiled = await WebAssembly.compile(buffer);
+  return compiled;
+};
+
+const load = async (fileName, importObject = {}, opts = { browser: false }) => {
+  const compiled = opts.browser
+    ? await compileInBrowserEnv(fileName)
+    : await compileInServerEnv(fileName);
```
Comment (tshemsedinov) [inline lib/node.js:70]:
```suggestion
  const compile = opts.browser ? compileInBrowserEnv : compileInServerEnv;
  const compiled = await compile(fileName);
```

**Review by tshemsedinov — APPROVED** [review]

**Comment by tshemsedinov** [comment]
Need to rebase on master @rustworthy

**Pull request merged**